Replaced API v1 with Cousteau; Fixed prepare.py, get-ips.py, measure.py

slowr opened the pull request:
Major Changes:

1. Measure, MeasurementFetch now uses Cousteau instead of urllib. ProbeInfo for archive is not supported by Cousteau so urllib2 still exists. MeasurementInfo unchanged - unused.

2. Fixed prepare.py where probe info data structure is different and the script was failing to gather nodes. More specific; probe status and probe geometry entries are different.

3. Fixed a problem where Measure.py was failing to gather source and destination probes because tag structure was changed to a list and the check was always false.

Minor Changes:

1. IPInfoCache had a minor typo causing findAsn func to return empty asn information.

2. Added threading on get-ips.py to improve the speed of this specific step

3. Change the indentation of some script from 3 spaces to 4 spaces to avoid problems with multiple IDEs

4. Added futures package to requirements file (used for parallelity)

emileaben commented:
tested, and now my prepare.py step fails with:

`Traceback (most recent call last):
  File "../../../prepare.py", line 491, in <module>
    sel_probes_for_cc = do_probe_selection( probes_cc, conf, basedata )
  File "../../../prepare.py", line 115, in do_probe_selection
    if status['id'] != 1:
TypeError: 'int' object has no attribute '__getitem__'`

slowr commented:
Weird. For me the prb_info['status'] is a <type 'dict'> and not an integer. What version of ripe cousteau  do you have? Also, there is a possibility that ProbeInfo.query that uses cousteau returns dictionary but ProbeInfo.query_archive returns an integer.

My version of cousteau:

Metadata-Version: 1.1
Name: ripe.atlas.cousteau
Version: 1.2

slowr commented:
As it seems. API v2 has changed the returned data structure and uses a dictionary for type, replaced latitude and longitude with a dictionary named geometry and changed tags to dictionary as well.

[API v2 Example](https://atlas.ripe.net/api/v2/probes/?latitude__gte=40.0&latitude__lte=46.0&longitude__gte=70&longitude__lte=75)
